### PR TITLE
yazi-unwrapped: build new yazi-cli tool

### PR DESCRIPTION
**Pull request by eljamm**
## Description of changes

See https://github.com/sxyazi/yazi/issues/914

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eljamm** [comment]
Result of `nixpkgs-review pr 321937` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>yazi</li>
    <li>yazi-unwrapped</li>
  </ul>
</details>

**Comment by eljamm** [comment]
Tested CLI tool a little bit following the yazi [DDS docs](https://yazi-rs.github.io/docs/dds/#cli) page and it seems to work.

## Testing steps
- Open yazi and enter a subshell
- Find out the instance ID using `echo $YAZI_ID`. You can exit the yazi subshell, now.
- In another shell, invoke a command to the instance using the cli. For example, to change the directory:
```
$ ./result/bin/ya pub "1719138555289021" dds-cd --str "/tmp"
```
- Back in the yazi instance, you'll notice that the working directory has changed

**Review by XYenon — CHANGES_REQUESTED** [review]
Yazi the wrapper also need to be changed to make `ya` available.

File: pkgs/by-name/ya/yazi-unwrapped/package.nix
Comment (XYenon) [inline pkgs/by-name/ya/yazi-unwrapped/package.nix:28]: I have added it to `default-members`, so we can remove this next version.

https://github.com/sxyazi/yazi/pull/944/files

Comment (eljamm) [reply pkgs/by-name/ya/yazi-unwrapped/package.nix:28]: So do you think we should just wait for the next release or do we add a message like this:

```nix
  # remove in the next release
  cargoBuildFlags = ["-p" "yazi-fm" "-p" "yazi-cli"];
```

**Comment by eljamm** [comment]
> Yazi the wrapper also need to be changed to make `ya` available.

Since the wrapper changed in [321076](https://github.com/NixOS/nixpkgs/pull/321076), I didn't want to modify it here until it was merged to avoid conflicts.

Would it have been better if I opened this as a draft, perhaps?

**Comment by eljamm** [comment]
The wrapper PR was merged so I've added the cli tool to the wrapper, tested again, and everything appears to be working.

I've also added a note to remove the cargo flags in the next release.

**Review by linsui — APPROVED** [review]

**Pull request merged**
